lib/vyasa/adapters/og_adapter.ex

```diff
@@ -27,15 +27,22 @@ defmodule Vyasa.Adapters.OgAdapter do
     System.tmp_dir() |> Path.join(filename)
   end
 
-  @doc """
-  Returns a string representing the filename for a particular verse from a particular chapter of
-  the gita. Currently just supports gita.
-  """
-  def encode_filename(src, [chapter_num, verse_num]) when src == :gita do
-    gita_uuid = "gita" # TODO: consider better unique ids based on what we want to support for our corpus.
-    gita_uuid <> "-" <> Integer.to_string(chapter_num) <> "-" <> Integer.to_string(verse_num) <> ".png"
-  end
+  # @doc """
+  # Returns a string representing the filename for a particular verse from a particular chapter of
+  # the gita. Currently just supports gita.
+  # """
+  # def encode_filename(src, [chapter_num, verse_num]) when src == :gita do
+  #   gita_uuid = "gita" # TODO: consider better unique ids based on what we want to support for our corpus.
+  #   gita_uuid <> "-" <> Integer.to_string(chapter_num) <> "-" <> Integer.to_string(verse_num) <> ".png"
+  # end
 
+  def encode_filename(module, params) do
+    param_delim = "~"
+    ext = ".png"
+    [module | params]
+    |> Enum.join(param_delim)
+    |> (&(&1 <> ext)).()
+  end
```

Comment:
I have often found path names more expressive using simple string concatenation 
```elixir
def encode_file_name(m_name, m) do 
"#{m_name}/#{uuid}#{unless is_nil(m.access),
          do: "/#{m.access}"}#{unless is_nil(m.count),
          do: "/#{m.count}"}#{unless is_nil(m.resolution),
          do: "/#{m.resolution}"}#{unless is_nil(m.height),
          do: "/#{m.height}x#{m.width}"}#{unless is_nil(m.ext),
          do: ".#{m.ext}"}"
end
```
This would hinge on a k-v approach to the arguments passed thru params than the current list based approach

Reply:
```